File: python/27.remove_element.py
from typing import List


class Solution:
    """
    https://leetcode.com/problems/remove-element
    """

    def removeElement(self, nums: List[int], val: int) -> int:
        try:
            while True:
                nums.remove(val)
        except ValueError:
            pass
        return len(nums)

File: cpp/27.remove_element.cpp
class Solution {
 public:
    int removeElement(std::vector<int>& nums, int val) {
        auto k = 0;
        auto i = 0;
        for (auto num : nums) {
            if (num != val) {
                nums[k] = nums[i];
                ++k;
            }
            ++i;
        }
        return k;
    }
};
